internal/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const defaultInterval = 10 * time.Second
+
+var logger log.LeveledLogger = log.NewFromGlobal(log.AddContext("pkg", "metrics"))
+
+// Config is base config for metrics
+type Config struct {
+	Publish bool
+}
+
+// IntervalConfig for interval collection
+type IntervalConfig struct {
+	Config
+	Interval time.Duration
+}
```

Comment:
`interval` and `publish` looks like fields of a `Config` struct
```suggestion
type Config struct {
	Publish bool
	Interval time.Duration
}
```

Reply:
Also I'm not sure we need a constructor that much, this is just a config struct. Maybe you can add a method on the configuration like `func (c *Config) SetDefaults()` to set defaults values for unset values.

Reply:
I like to keep creation of types to be centralised within the package.  I've had cases in larger codebases, where there were regressions after adding attributes to the types, but not updating every instantiation of that type outside of the package.

Reply:
I agree, although a `SetDefaults()` method would achieve the same thing. You can call it to the constructor where that config struct is passed to? As in, I guess it's the 'same difference' 😄